spring-cloud-task-batch/src/main/java/org/springframework/cloud/task/batch/partition/SimpleEnvironmentVariablesProvider.java

```diff
@@ -57,11 +59,19 @@ public void setEnvironmentProperties(Map<String, String> environmentProperties)
 		this.environmentProperties = environmentProperties;
 	}
 
+	public void setIncludeCurrentEnvironment(boolean includeCurrentEnvironment) {
```

Comment:
Needs JavaDoc.  Also the class level javadoc requires an update to state that its optional to include all current environment.  Fix this on merge